avssync/metrics.go

```diff
@@ -35,12 +36,32 @@ var (
 	}, []string{"quorum"})
 )
 
-func StartMetricsServer(metricsAddr string) {
+type Metrics struct {
```

Comment:
See my below comment. Feel like we should rename this to MetricsServer. It should just take a registry as argument but not register the prometheus variables into it (since its just a MetricsServer). Ideally this metricsServer would live in sdk.

Then you would first call the `registerPromVariables(reg prometheus.Registry)` in main to register the metrics, and then start the webserver (if needed, maybe put that as flag in the binary).